pallets/gear-builtin-actor/src/benchmarking.rs

```diff
@@ -0,0 +1,133 @@
+use frame_benchmarking::{benchmarks, impl_benchmark_test_suite};
+use gear_core::{
+    ids::{BuiltinId, ProgramId},
+    message::{DispatchKind, StoredDispatch, StoredMessage},
+};
+
+macro_rules! impl_builtin_actor {
+    ($name: ident, $id: literal) => {
+        pub struct $name {}
+
+        impl BuiltinActor<SimpleBuiltinMessage, u64> for $name {
+            fn handle(
+                _message: &SimpleBuiltinMessage,
+            ) -> (Result<Vec<u8>, BuiltinActorError>, u64) {
+                (Ok(Default::default()), Default::default())
+            }
+
+            fn max_gas_cost(_builtin_id: BuiltinId) -> u64 {
+                Default::default()
+            }
+        }
+        impl RegisteredBuiltinActor<SimpleBuiltinMessage, u64> for $name {
+            const ID: BuiltinId = BuiltinId($id as u64);
+        }
+    };
+}
+
+impl_builtin_actor!(DummyActor0, 0);
+impl_builtin_actor!(DummyActor1, 1);
+impl_builtin_actor!(DummyActor2, 2);
+impl_builtin_actor!(DummyActor3, 3);
+impl_builtin_actor!(DummyActor4, 4);
+impl_builtin_actor!(DummyActor5, 5);
+impl_builtin_actor!(DummyActor6, 6);
+impl_builtin_actor!(DummyActor7, 7);
+impl_builtin_actor!(DummyActor8, 8);
+impl_builtin_actor!(DummyActor9, 9);
+impl_builtin_actor!(DummyActor10, 10);
+
+benchmarks! {
+    lookup {
+        // Populate the storage with maximum (16, as of today) builtin actors ids
+        for i in 0_u64..16 {
+            let builtin_id = BuiltinId(i);
+            let actor_id = Pallet::<T>::generate_actor_id(builtin_id);
+            Actors::<T>::insert(actor_id, builtin_id);
+        }
+        let actor_id = ProgramId::from(100_u64);
+    }: {
+        BuiltinActorPallet::<T>::lookup(&actor_id)
+    } verify {
+        // No changes in runtime are expected since the actual dispatch doesn't take place.
+    }
+
+    calculate_id {
+        let builtin_id = BuiltinId(100_u64);
+    }: {
+        Pallet::<T>::generate_actor_id(builtin_id)
+    } verify {
+        // No changes in runtime are expected since the actual dispatch doesn't take place.
+    }
+
+    base_handle_weight {
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor0, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor1, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor2, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor3, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor4, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor5, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor6, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor7, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor8, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor9, _, _>();
+        let _ = BuiltinActorPallet::<T>::register_actor::<DummyActor10, _, _>();
```

Comment:
`handle` will not work with regiustered built-in actors as it takes `BuiltinId` as the input and therefore don't perform lookup

Reply:
We are not benchmarking `lookup` with this test. The reason for registering a bunch of actors is to populate the storage to simulate iteration through a tuple elements that might be taking place in reality.
This test is rather useless anyway because with 16 actors it takes literally 0 weight to identify the type whose `handle` should be called. Keeping it exclusively for the sake of generality.

Reply:
Yes, I mean that `lookup` isn't executed, so you don't need to fill storage as it'll be not queried. What actually will be used are 2 of built-in actors you registered in mock

Reply:
Anyway, all this doesn't matter anymore, because in the latest version of the framework we completely outed out from storing known builtin ids in favour of deriving them in every block on the fly (which is at least 5 times cheaper based on our benchmarks comparing to fetching values from storage).